<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Olga Amelkina" />


<title>Part 1. Visualization of gene expression patterns for all diet types</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Project_BFF_sperm_transcriptomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/OlgaAmelkina/Project_BFF_sperm_transcriptomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Part 1. Visualization of gene expression patterns for all diet types</h1>
<h4 class="author">Olga Amelkina</h4>
<h4 class="date">3/23/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Project_BFF_sperm_transcriptomics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220323code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220323)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220323code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220323)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOlgaAmelkinaProjectBFFspermtranscriptomicstreeaf2f40eae07c5a45954527e4dd6e1a90c6d96be3targetblankaf2f40ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/OlgaAmelkina/Project_BFF_sperm_transcriptomics/tree/af2f40eae07c5a45954527e4dd6e1a90c6d96be3" target="_blank">af2f40e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomOlgaAmelkinaProjectBFFspermtranscriptomicstreeaf2f40eae07c5a45954527e4dd6e1a90c6d96be3targetblankaf2f40ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/OlgaAmelkina/Project_BFF_sperm_transcriptomics/tree/af2f40eae07c5a45954527e4dd6e1a90c6d96be3" target="_blank">af2f40e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  Part1_Data_Visualization.Rmd
    Untracked:  Part1_Data_Visualization.html
    Untracked:  data/Mnigripes_GeneIDs.csv
    Untracked:  data/Plots/
    Untracked:  data/RSEM/
    Untracked:  data/metadata.csv
    Untracked:  data/metadata_3groups.csv
    Untracked:  data/metadata_3groups_excl2.csv
    Untracked:  data/metadata_excl2.csv
    Untracked:  data/metadata_exclVar.csv
    Untracked:  renv.lock
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Part1_Data_Visualization.Rmd</code>) and HTML (<code>docs/Part1_Data_Visualization.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/OlgaAmelkina/Project_BFF_sperm_transcriptomics/blob/af2f40eae07c5a45954527e4dd6e1a90c6d96be3/analysis/Part1_Data_Visualization.Rmd" target="_blank">af2f40e</a>
</td>
<td>
Olga Amelkina
</td>
<td>
2022-03-23
</td>
<td>
Part 1.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="groups-used-for-analysis" class="section level1">
<h1>1.Groups used for analysis</h1>
<ol style="list-style-type: decimal">
<li><strong>Control</strong> diet, labeled C01-C10, n = 10<br />
</li>
<li><strong>Vitamin E</strong> diet, labeled VE1.1-2 for 1st generation (n = 2) and VE2.01-13 for 2d generation (n = 13)<br />
</li>
<li><strong>Wild</strong> individuals, labeled W1-4, n = 4<br />
</li>
<li><strong>Carcass</strong> diet, labeled Car1-2, n = 2<br />
</li>
<li><strong>Various</strong> individuals excluded from the study, labeled Var1-2, n = 2</li>
</ol>
<div id="list-of-all-individuals" class="section level2">
<h2>1.1.List of all individuals</h2>
<table>
<caption>Metadata</caption>
<thead>
<tr class="header">
<th align="left">ID</th>
<th align="left">Diet</th>
<th align="left">Seq_ID</th>
<th align="left">Animal_ID</th>
<th align="left">Name</th>
<th align="left">Age</th>
<th align="left">Location</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C01</td>
<td align="left">Control</td>
<td align="left">NA-002</td>
<td align="left">9571</td>
<td align="left">Arthur</td>
<td align="left">2y7m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">C02</td>
<td align="left">Control</td>
<td align="left">NA-003</td>
<td align="left">9592</td>
<td align="left">Calo</td>
<td align="left">2y7m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">C03</td>
<td align="left">Control</td>
<td align="left">NA-007</td>
<td align="left">9519</td>
<td align="left">Bannner</td>
<td align="left">2y9m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">C04</td>
<td align="left">Control</td>
<td align="left">NA-009</td>
<td align="left">9430</td>
<td align="left">Cushman</td>
<td align="left">2y7m</td>
<td align="left">LZG</td>
</tr>
<tr class="odd">
<td align="left">C05</td>
<td align="left">Control</td>
<td align="left">NA-014</td>
<td align="left">9350</td>
<td align="left">Boatman</td>
<td align="left">2y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">C06</td>
<td align="left">Control</td>
<td align="left">NA-016</td>
<td align="left">9128</td>
<td align="left">Hawksbill</td>
<td align="left">3y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">C07</td>
<td align="left">Control</td>
<td align="left">NA-022</td>
<td align="left">9276</td>
<td align="left">Wilde</td>
<td align="left">3y7m</td>
<td align="left">RMA</td>
</tr>
<tr class="even">
<td align="left">C08</td>
<td align="left">Control</td>
<td align="left">NA-023</td>
<td align="left">9291</td>
<td align="left">Kibosh</td>
<td align="left">2y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">C09</td>
<td align="left">Control</td>
<td align="left">NA-026</td>
<td align="left">9443</td>
<td align="left">Meusli</td>
<td align="left">2y7m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">C10</td>
<td align="left">Control</td>
<td align="left">NA-029</td>
<td align="left">9576</td>
<td align="left">Pikachu</td>
<td align="left">3y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">Car1</td>
<td align="left">Carcass</td>
<td align="left">NA-001</td>
<td align="left">9469</td>
<td align="left">Doublemint</td>
<td align="left">2y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">Car2</td>
<td align="left">Carcass</td>
<td align="left">NA-027</td>
<td align="left">9452</td>
<td align="left">Rockbiter</td>
<td align="left">2y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">Var1</td>
<td align="left">Various</td>
<td align="left">NA-004</td>
<td align="left">9051</td>
<td align="left">Tater</td>
<td align="left">4y7m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">Var2</td>
<td align="left">Various</td>
<td align="left">NA-015</td>
<td align="left">9072</td>
<td align="left">Woodford</td>
<td align="left">4y7m</td>
<td align="left">RMA</td>
</tr>
<tr class="odd">
<td align="left">VE1.1</td>
<td align="left">1st.gen.VitE</td>
<td align="left">NA-005</td>
<td align="left">8995</td>
<td align="left">CaptainBarnacles</td>
<td align="left">4y7m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">VE1.2</td>
<td align="left">1st.gen.VitE</td>
<td align="left">NA-006</td>
<td align="left">8593</td>
<td align="left">Dash</td>
<td align="left">5y9m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">VE2.01</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-008</td>
<td align="left">9227</td>
<td align="left">Balthasar</td>
<td align="left">3y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">VE2.02</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-010</td>
<td align="left">9247</td>
<td align="left">Bosler</td>
<td align="left">3y8m</td>
<td align="left">LZG</td>
</tr>
<tr class="odd">
<td align="left">VE2.03</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-011</td>
<td align="left">9219</td>
<td align="left">Cameron</td>
<td align="left">3y8m</td>
<td align="left">LZG</td>
</tr>
<tr class="even">
<td align="left">VE2.04</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-012</td>
<td align="left">9476</td>
<td align="left">Flatback</td>
<td align="left">2y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">VE2.05</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-013</td>
<td align="left">9311</td>
<td align="left">HopperStoppeer</td>
<td align="left">3y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">VE2.06</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-017</td>
<td align="left">9135</td>
<td align="left">Heavenly</td>
<td align="left">3y7m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">VE2.07</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-018</td>
<td align="left">9150</td>
<td align="left">WillFerret</td>
<td align="left">2y9m</td>
<td align="left">RMA</td>
</tr>
<tr class="even">
<td align="left">VE2.08</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-019</td>
<td align="left">9181</td>
<td align="left">Oslo</td>
<td align="left">3y7m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">VE2.09</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-020</td>
<td align="left">9194</td>
<td align="left">Statler</td>
<td align="left">3y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">VE2.10</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-021</td>
<td align="left">9195</td>
<td align="left">Waldorf</td>
<td align="left">3y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">VE2.11</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-024</td>
<td align="left">9318</td>
<td align="left">Myrcenary</td>
<td align="left">3y8m</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">VE2.12</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-025</td>
<td align="left">9373</td>
<td align="left">Martini</td>
<td align="left">2y7m</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">VE2.13</td>
<td align="left">2d.gen.VitE</td>
<td align="left">NA-028</td>
<td align="left">9568</td>
<td align="left">Wellington</td>
<td align="left">3y8m</td>
<td align="left">RMA</td>
</tr>
<tr class="even">
<td align="left">W1</td>
<td align="left">Wild</td>
<td align="left">NA-030</td>
<td align="left">RMA13</td>
<td align="left">RMA13</td>
<td align="left">NA</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">W2</td>
<td align="left">Wild</td>
<td align="left">NA-031</td>
<td align="left">RMA14</td>
<td align="left">RMA14</td>
<td align="left">NA</td>
<td align="left">FCC</td>
</tr>
<tr class="even">
<td align="left">W3</td>
<td align="left">Wild</td>
<td align="left">NA-032</td>
<td align="left">RMA15</td>
<td align="left">RMA15</td>
<td align="left">NA</td>
<td align="left">FCC</td>
</tr>
<tr class="odd">
<td align="left">W4</td>
<td align="left">Wild</td>
<td align="left">NA-033</td>
<td align="left">RMA16</td>
<td align="left">RMA16</td>
<td align="left">NA</td>
<td align="left">FCC</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="code-for-getting-normalized-expression-data-with-deseq2" class="section level1">
<h1>2.Code for getting normalized expression data with DESeq2</h1>
<p>This code is hidden from the report due to its length. If needed, it can be found in Part1_Data_Visualization.Rmd file.</p>
</div>
<div id="principal-component-analysis" class="section level1">
<h1>3.Principal Component Analysis</h1>
<div id="pca-plot-for-all-individuals" class="section level2">
<h2>3.1.PCA plot for all individuals</h2>
<div id="unlabeled" class="section level4">
<h4>Unlabeled</h4>
<p><img src="figure/Part1_Data_Visualization.Rmd/PCA-all-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="labeled" class="section level4">
<h4>Labeled</h4>
<p><img src="figure/Part1_Data_Visualization.Rmd/PCA-all-labeled-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>From these PCA plots we can see that there is no clear grouping of individuals based on transcriptome-wide normalized gene expression. Two individuals seem to be separated from the rest: one from Control diet (C09) and one Wild (W1). All this might implicate that the differences between individuals are bigger than between diets.</p>
</div>
</div>
<div id="pca-plots-excluding-c09-and-w1" class="section level2">
<h2>3.2.PCA plots excluding C09 and W1</h2>
<p>Exclusion of these two individuals did not change the observed pattern (absence of grouping).</p>
<div id="unlabeled-1" class="section level4">
<h4>Unlabeled</h4>
<p><img src="figure/Part1_Data_Visualization.Rmd/PCA-excl2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="labeled-1" class="section level4">
<h4>Labeled</h4>
<p><img src="figure/Part1_Data_Visualization.Rmd/PCA-excl2-labeled-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="pca-plots-for-pc3-6" class="section level2">
<h2>3.3.PCA plots for PC3-6</h2>
<p>Next PCA plots are created to show additional principal components. However, still no grouping pattern can be observed.</p>
<div id="pc3-and-pc4" class="section level3">
<h3>PC3 and PC4</h3>
<p><img src="figure/Part1_Data_Visualization.Rmd/PCA-PC3-PC4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pc5-and-pc6" class="section level3">
<h3>PC5 and PC6</h3>
<p><img src="figure/Part1_Data_Visualization.Rmd/PCA-PC5-PC6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="heatmap-with-hierarchical-clustering" class="section level1">
<h1>4.Heatmap with hierarchical clustering</h1>
<div id="heatmap-for-all-indibiduals" class="section level2">
<h2>4.1.Heatmap for all indibiduals</h2>
<p>Heatmap with one-way hierarchical clustering analysis (Euclidean method, complete linkage) using Z-score for RLE normalized values of all expressed genes.<br />
Labels correspond to individual IDs from the Metadata table. <img src="figure/Part1_Data_Visualization.Rmd/Heatmap-all-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="graphs-used-for-nadyas-dissertation" class="section level1">
<h1>5.Graphs used for Nadya’s dissertation</h1>
<p>For Nadya’s dissertation, we decided to exclude Various group (individuals excluded from the study) and unite 2 generations of Vitamin E diet into one group (1st generation has only 2 individuals)</p>
<div id="pca-plot" class="section level2">
<h2>5.1.PCA plot</h2>
<div id="unlabeled-2" class="section level4">
<h4>Unlabeled</h4>
<p><img src="figure/Part1_Data_Visualization.Rmd/PCA-Figure-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="labeled-2" class="section level4">
<h4>Labeled</h4>
<p><img src="figure/Part1_Data_Visualization.Rmd/PCA-Figure-labeled-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="heatmap" class="section level2">
<h2>5.2. Heatmap</h2>
<p><img src="figure/Part1_Data_Visualization.Rmd/Heatmap-Figure-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    
system code page: 65001

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] pheatmap_1.0.12 ggplot2_3.3.5   knitr_1.37      workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8         RColorBrewer_1.1-2 highr_0.9          jquerylib_0.1.4   
 [5] compiler_4.1.0     pillar_1.7.0       later_1.3.0        git2r_0.30.1      
 [9] tools_4.1.0        getPass_0.2-2      digest_0.6.29      gtable_0.3.0      
[13] evaluate_0.15      tibble_3.1.6       lifecycle_1.0.1    pkgconfig_2.0.3   
[17] rlang_1.0.1        cli_3.1.1          rstudioapi_0.13    yaml_2.2.2        
[21] xfun_0.29          fastmap_1.1.0      withr_2.4.3        httr_1.4.2        
[25] stringr_1.4.0      fs_1.5.2           vctrs_0.3.8        grid_4.1.0        
[29] rprojroot_2.0.2    glue_1.6.1         R6_2.5.1           processx_3.5.2    
[33] fansi_1.0.2        rmarkdown_2.11     farver_2.1.0       callr_3.7.0       
[37] magrittr_2.0.2     whisker_0.4        scales_1.1.1       ps_1.6.0          
[41] promises_1.2.0.1   htmltools_0.5.2    ellipsis_0.3.2     colorspace_2.0-2  
[45] renv_0.15.4        httpuv_1.6.5       labeling_0.4.2     utf8_1.2.2        
[49] stringi_1.7.6      munsell_0.5.0      crayon_1.5.0      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
